src/components/admin/PersonLookup.vue

```diff
@@ -45,29 +51,42 @@ export default {
   }),
   watch: {
     search(snippet) {
+      this.debouncedSearch(snippet)
+    },
+    selected(suggestion) {
+      if (suggestion) {
+        this.onSelectResult(suggestion.value)
+      }
+    }
+  },
+  methods: {
+    executeSearch(snippet) {
       if (snippet) {
         this.isSearching = true
-        searchUsers(snippet, this.excludeUids).then(results => {
+        const apiSearch = this.instructorLookup ? searchInstructors : searchUsers
+        apiSearch(snippet, this.excludeUids).then(results => {
           this.suggestions = this.$_.map(results, this.suggest)
           this.isSearching = false
         })
       } else {
         this.suggestions = []
       }
     },
-    selected(suggestion) {
-      if (suggestion) {
-        this.onSelectResult(suggestion.value)
-      }
-    }
-  },
-  methods: {
     suggest(user) {
       return {
         text: `${user.firstName} ${user.lastName} (${user.uid})`,
         value: user
       }
     }
+  },
+  created() {
+    this.debouncedSearch = this.$_.debounce(this.executeSearch, 300)
```

Comment:
Performance on the autocomplete seems iffy and I'm not yet sure why, but adding a lodash debounce to reduce the number of piled-on API calls seems like a good first step.